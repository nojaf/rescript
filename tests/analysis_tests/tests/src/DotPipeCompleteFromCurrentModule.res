module X = {
    type t
}

module Y = {
    open X

    let z = (x: t) => ""

    let a = (x:t) => {
        // x.
        //   ^com
        ()
    }

    let b = (x:t) => 4
}


module Types = {
    type comp
    type context
    type vec2
}

module PosComp = (
  T: {
    type t
  },
) => {
  open Types

  @send
  external addPos: (context, float, float) => comp = "pos"

  @send
  external addPosFromVec2: (context, vec2) => comp = "pos"
}

module SpriteComp = (
    T: {
        type t
    }
)
 => {
    open Types

    @send
    external addSprite: (context, string) => comp = "sprite"
 }

external k: Types.context = "k"

module Wall = {
    type t

    include PosComp({ type t = t })

    let blah = (k: Types.context) => ""

    let make = () => {
        [
            // k.
            //   ^com

            // add
            //    ^com
        ]
    }

    module Poster = {
        type t

        include SpriteComp({ type t = t })

        let make = () => {
            [
                // k.
                //   ^com
            ]
        }
    }
}