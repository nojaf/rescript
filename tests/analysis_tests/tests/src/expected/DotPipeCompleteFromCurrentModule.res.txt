Complete src/DotPipeCompleteFromCurrentModule.res 10:13
posCursor:[10:13] posNoWhite:[10:12] Found expr:[9:12->13:5]
posCursor:[10:13] posNoWhite:[10:12] Found expr:[10:11->12:10]
posCursor:[10:13] posNoWhite:[10:12] Found expr:[10:11->10:13]
Pexp_field [10:11->10:12] _:[12:8->10:13]
Completable: Cpath Value[x].""
Raw opens: 1 X.place holder
Package opens Stdlib.place holder Pervasives.JsxModules.place holder
Resolved opens 2 Stdlib DotPipeCompleteFromCurrentModule
ContextPath Value[x].""
ContextPath Value[x]
Path x
ContextPath Value[x]->
ContextPath Value[x]
Path x
CPPipe pathFromEnv:X found:true
Path X.
Path 
[{
    "label": "->z",
    "kind": 12,
    "tags": [],
    "detail": "X.t => string",
    "documentation": null,
    "sortText": "z",
    "insertText": "->z",
    "additionalTextEdits": [{
      "range": {"start": {"line": 10, "character": 12}, "end": {"line": 10, "character": 13}},
      "newText": ""
      }]
  }]

Complete src/DotPipeCompleteFromCurrentModule.res 60:17
posCursor:[60:17] posNoWhite:[60:16] Found expr:[58:15->63:5]
posCursor:[60:17] posNoWhite:[60:16] Found expr:[59:8->62:9]
posCursor:[60:17] posNoWhite:[60:16] Found expr:[60:15->60:17]
Pexp_field [60:15->60:16] _:[62:8->60:17]
Completable: Cpath Value[k].""
Package opens Stdlib.place holder Pervasives.JsxModules.place holder
Resolved opens 1 Stdlib
ContextPath Value[k].""
ContextPath Value[k]
Path k
ContextPath Value[k]->
ContextPath Value[k]
Path k
CPPipe pathFromEnv:Types found:true
Path Types.
Path 
[{
    "label": "->addPosFromVec2",
    "kind": 12,
    "tags": [],
    "detail": "(Types.context, Types.vec2) => Types.comp",
    "documentation": null,
    "sortText": "addPosFromVec2",
    "insertText": "->addPosFromVec2",
    "additionalTextEdits": [{
      "range": {"start": {"line": 60, "character": 16}, "end": {"line": 60, "character": 17}},
      "newText": ""
      }]
  }, {
    "label": "->addPos",
    "kind": 12,
    "tags": [],
    "detail": "(Types.context, float, float) => Types.comp",
    "documentation": null,
    "sortText": "addPos",
    "insertText": "->addPos",
    "additionalTextEdits": [{
      "range": {"start": {"line": 60, "character": 16}, "end": {"line": 60, "character": 17}},
      "newText": ""
      }]
  }]

Complete src/DotPipeCompleteFromCurrentModule.res 72:21
posCursor:[72:21] posNoWhite:[72:20] Found expr:[70:19->75:9]
posCursor:[72:21] posNoWhite:[72:20] Found expr:[71:12->74:13]
posCursor:[72:21] posNoWhite:[72:20] Found expr:[72:19->72:21]
Pexp_field [72:19->72:20] _:[74:12->72:21]
Completable: Cpath Value[k].""
Package opens Stdlib.place holder Pervasives.JsxModules.place holder
Resolved opens 1 Stdlib
ContextPath Value[k].""
ContextPath Value[k]
Path k
ContextPath Value[k]->
ContextPath Value[k]
Path k
CPPipe pathFromEnv:Types found:true
Path Types.
Path 
[{
    "label": "->addSprite",
    "kind": 12,
    "tags": [],
    "detail": "(Types.context, string) => Types.comp",
    "documentation": null,
    "sortText": "addSprite",
    "insertText": "->addSprite",
    "additionalTextEdits": [{
      "range": {"start": {"line": 72, "character": 20}, "end": {"line": 72, "character": 21}},
      "newText": ""
      }]
  }]

